java/README.md

```diff
@@ -36,3 +42,39 @@ Parancssori futtatás fordítás után Windowson:
 2. A projekt könyvtárában a parancsértelmezőjével futtassa a következő parancsot:
 java -cp out\production\java hu.holyoil.Main
 3. Alternatív: futtassa a run.bat fájlt, amely ugyanazt a parancsot futtatja le, amelyet fent is említettünk.
+
+## Tesztelés
+
+A testdir mappában test_*.txt fájlnévvel léteznek a dokumentált teszteseteink, ahol a * a teszteset megnevezését referálja.
+Ugyanezen mappában res_*.txt fájlnévvel léteznek a dokumentált tesztesetek várt eredményei.
+
+### Teszt kimenetek legyártása
+
+Parancssorban Windowson (fordítás után), Python3 mellett:
+python3 run_all.py
+
+### Teszt kimenetek legyártása majd összehasonlítása elvárttal
+
+Parancssorban Windowson (fordítás után):
+
+1. Feltételezve van, hogy a projekt könyvtárában megtörtént a fordítás (lásd feljebb).
+2. Futtassa a test_all.bat fájlt, amely lefordítja és futtatja a teszt-komparáló segédprogramot.
+
+### Kézzel írt teszt kimenete
+
+Parancssorban Windowson (fordítás után):
+
+0. Írja meg a tesztet a testdir mappában test_*.txt fájlnévvel, ahol a * helyére mehet a teszteset megnevezése ízlés szerint.
+1. Feltételezve van, hogy a projekt könyvtárában megtörtént a fordítás (lásd feljebb).
+2. A projekt könyvtárában a parancsértelmezőjével futtassa a következő parancsot (a * helyére az elkészült tesztfájl megnevezése kerül):
+run.bat < test_*.txt
```

Comment:
Shouldn't this be `testdir/test_*.txt`? `test_*.txt` would mean that the file is in the same folder as `run.bat`.